osidb/tests/test_flaw.py

```diff
@@ -1228,3 +1241,72 @@ def test_validate_exceptional_affectedness_resolution(
         assert should_raise == bool("flaw_exceptional_affect_status" in affect._alerts)
+
+    @pytest.mark.parametrize("entity", ["flaw", "affect"])
+    @pytest.mark.parametrize(
+        "affectedness,resolution,should_raise",
+        [
+            (Affect.AffectAffectedness.NEW, Affect.AffectResolution.NOVALUE, False),
+            (Affect.AffectAffectedness.NEW, Affect.AffectResolution.DEFER, False),
+            (Affect.AffectAffectedness.NEW, Affect.AffectResolution.WONTFIX, False),
+            (Affect.AffectAffectedness.NEW, Affect.AffectResolution.OOSS, False),
+            (Affect.AffectAffectedness.AFFECTED, Affect.AffectResolution.FIX, False),
+            (Affect.AffectAffectedness.AFFECTED, Affect.AffectResolution.DEFER, False),
+            (
+                Affect.AffectAffectedness.AFFECTED,
+                Affect.AffectResolution.DELEGATED,
+                False,
+            ),
+            (
+                Affect.AffectAffectedness.AFFECTED,
+                Affect.AffectResolution.WONTREPORT,
+                False,
+            ),
+            (
+                Affect.AffectAffectedness.AFFECTED,
+                Affect.AffectResolution.WONTFIX,
+                False,
+            ),
+            (Affect.AffectAffectedness.AFFECTED, Affect.AffectResolution.OOSS, False),
+            (
+                Affect.AffectAffectedness.NOTAFFECTED,
+                Affect.AffectResolution.NOVALUE,
+                False,
+            ),
+            (Affect.AffectAffectedness.NEW, Affect.AffectResolution.FIX, True),
+            (Affect.AffectAffectedness.AFFECTED, Affect.AffectResolution.NOVALUE, True),
+            (
+                Affect.AffectAffectedness.NOTAFFECTED,
+                Affect.AffectResolution.DEFER,
+                True,
+            ),
+        ],
+    )
+    def test_validate_flaw_affects_status_resolution(
+        self, entity, affectedness, resolution, should_raise
+    ):
+        """
+        Test that error is raised if any flaw's affect have a
+        invalid combination of affectedness and resolution
+        """
+        flaw = FlawFactory()
+        affect = AffectFactory.build(
+            flaw=flaw, affectedness=affectedness, resolution=resolution
+        )
+        affect.save(raise_validation_error=False)
+        if should_raise:
+            if entity == "flaw":
+                with pytest.raises(
+                    ValidationError,
+                    match=f"{affect.resolution} is not a valid resolution for {affect.affectedness}.",
+                ):
+                    flaw.save()
+            else:
+                with pytest.raises(
+                    ValidationError,
+                    match=f"{affect.resolution} is not a valid resolution for {affect.affectedness}.",
+                ):
+                    affect.save()
+        else:
+            assert affect.save() is None
+            assert flaw.save() is None
```

Comment:
Added the suggestion, thank you.

Reply:
Second thoughts on that: I will remove the validation on flaw side because it only interacts with affect API making this not necessary anymore. Sorry for the extra work helping me with this.